### PR TITLE
add theme-lint, remove some gems

### PR DESCRIPTION
**Pull request by NathanPJF**
No immediate need to merge, but getting this started.

**Why**:  Move theme i18n linting to the package.json so folks don't need circle Ci testing / ruby tests.

Other notes:
* Good candidate for hooks.  Run a linting task post build.

To do:
* Make Circle Ci happy
* Thorough check of Jekyll build for docs

### CODE REVIEW

**Comment by NathanPJF** [comment]
Note: Add a command for `npm run tests` that will run lint and can be expanded to run more test in the future.

**Comment by chrisberthe** [comment]
Related topic https://github.com/Shopify/slate-tools/issues/5. Might be better to integrate this into `slate-tools` directly?

**Comment by NathanPJF** [comment]
@chrisberthe I'm down with the idea of integrating theme linting into one of the package.  It will be linting the `dist/` directory.  Does slate-tools sounds like the best spot?  Or should it be a core `slate` command?

Maybe slate-tools and it can be an optional pre-deploy / post-build task?

Let's try it out as slate-tools, and if it works out we'll modify this PR accordingly.

File: circle.yml [hunk 1]
```diff
@@ -18,5 +19,7 @@ dependencies:
     - .bundle/
 
 test:
+  pre:
+    - slate build
   override:
-    - bundle exec rspec spec
+    - npm run lint
```
Comment (chrisberthe) [inline circle.yml]: "slate test"

File: package.json [hunk 2]
```diff
@@ -8,8 +8,12 @@
     "type": "git",
     "url": "https://github.com/Shopify/slate.git"
   },
+  "scripts": {
+    "lint": "theme-lint ./dist"
+  },
   "devDependencies": {
     "@shopify/slate-tools": "0.2.7",
+    "@shopify/theme-lint": "^1.0.2",
```
Comment (chrisberthe) [inline package.json]: Can remove this and the "lint" script above.

**Comment by NathanPJF** [comment]
@chrisberthe want to approve your own changes? 😜 

They look good to me ✅

**Pull request merged**